x/posts/spec/02-concepts.md

```diff
@@ -0,0 +1,202 @@
+| `POST_REFERENCE_TYPE_UNSPECIFIED` | No reference specified                                   |
+| `POST_REFERENCE_TYPE_REPLY`       | This reference represents a reply to the specified post  |
+| `POST_REFERENCE_TYPE_QUOTE`       | This reference represents a quote of the specified post  |
+| `POST_REFERENCE_TYPE_REPOST`      | This reference represents a repost of the specified post |
+
+##### Post ID
+The ID of the referenced post.
+
+##### Position
+This field should be used only when the `Type` field is equal to `TYPE_QUOTE` to indicates where the `PostReference`
+start in the post's text.
+
+### Reply Setting
+This field contains the possible reply settings that a post can have. It can be one of the following values:
+
+| **Name**                    | **Description**                                              |  
+|:----------------------------|:-------------------------------------------------------------|
+| `REPLY_SETTING_UNSPECIFIED` | No reference specified                                       |
+| `REPLY_SETTING_EVERYONE`    | This reference represents a reply to the specified post      |
+| `REPLY_SETTING_FOLLOWERS`   | This reference represents a quote of the specified post      |
+| `REPLY_SETTING_MUTUAL`      | This reference represents a repost of the specified post     |
+| `REPLY_SETTING_MENTIONS`    | Only people mentioned inside this post will be able to reply |
+
+### Creation Date
+The creation date of the post.
+
+### Last Edited Date
+The las time the post has been edited.
+
+## Attachment
+An attachment represent something that can be added to a post in order to enrich it or give it some additional use cases.
+
+### Subspace ID
+The [subspace] ID indicates the ID of the Dapp where the attachment is hosted and lives.
+
+### Post ID
+The [post](#Post) ID to which the attachment is linked.
+
+### ID
+The unique ID that identifies the attachment. This ID is automatically assigned the same way the post one is.
+
+### Content
+The content of the attachment. It can be either:
+- A media content (pictures, clips, videos, GIFs);
+- A poll.
+
+## Media
+The Media structure represent a media content of any kind from pictures, to clips, videos, GIFs, etc... attached to 
+a post.
+
+### URI
+The URI address referencing the media.
+
+### Mime type
+The [mime type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types) of the media.
+
+## Poll
+The Poll structure represents a poll attached to a post.
+
+### Question
+The question field define the question of the poll.
+
+### Provided answers
+The possible answers choices the users have to reply the poll. They are represented by a `ProvidedAnswer` structure.
+
+#### Provided Answer
+The representation of a provided answer for a poll.
+
+##### Text (Optional)
+The text of the answer.
+
+##### Attachments (Optional)
+The [attachments](#attachment) of the answer. If not provided, a text answer has to be specified.
```

Comment:
```suggestion
The [attachments](#attachment) associated to the answer. This can be useful to allow polls where answers are represented by a media content (e.g. an image) instead of a text. Note that either one of the text or attachments field must be specified within a provided answer to make sure it's valid.
```